### ISSUE TITLE
Handle not intialized notmuch database better

### ISSUE THREAD

**Issue by rumpelsepp** [astroidmail/astroid#63]
From a fresh install:

```
$ astroid
[info ] (12:35:25): welcome to astroid! - 0.4.r209.g9a1f7f8
[debug] (12:35:25): utf8: 1, UTF-8
[debug] (12:35:25): loading gtk..
[info ] (12:35:25): cf: loading: "/home/stefan/.config/astroid/config"
[info ] (12:35:25): cf: version: 2
[warn ] (12:35:25): cf: missing values in config have been updated with defaults (old version: 2, new: 2)
[warn ] (12:35:25): cf: writing back config to: "/home/stefan/.config/astroid/config"
[info ] (12:35:25): notmuch db: ~/.mail
[info ] (12:35:25): date: init.
[info ] (12:35:25): ac: initializing accounts..
[info ] (12:35:25): ac: setup account: charlie for Charlie Root (default: 1)
[info ] (12:35:25): global actions: set up.
[info ] (12:35:25): poll: setting up.
[debug] (12:35:25): poll: interval: 60
[debug] (12:35:25): poll: requested..
[warn ] (12:35:25): astroid: starting a new window..
[info ] (12:35:25): poll: polling: /home/stefan/.config/astroid/poll.sh
[info ] (12:35:25): db: open db read-only.
Error opening database at /home/stefan/.mail/.notmuch: No such file or directory
[ERROR] (12:35:25): db: error: failed opening database for reading, have you configured the notmuch database path correctly?

(astroid:24404): glibmm-ERROR **:
unhandled exception (type std::exception) in signal handler:
what: failed to open database (read-only)

[1]    24404 trace trap (core dumped)  astroid
```

**Comment by ff2000** [comment]
I see two options:
1) Do not crash, but show the user a dialog box saying "Database not configured, RTFM and restart astroid"
2) Have a look at ~/.notmuch-config, read the db-path from there and ask the user if it's OK to set that in the config. I'm against automatically setting the DB path, because probably the user just wants to test astroid and therefore has a dedicated testing-environment.

**Comment by rumpelsepp** [comment]
> Do not crash, but show the user a dialog box saying "Database not configured, RTFM and restart astroid"

I would even be fine with this.

**Comment by gauteh** [comment]
Stefan Tatschner writes on January 17, 2016 13:04:
>> Do not crash, but show the user a dialog box saying "Database not configured, RTFM and restart astroid"
> 
> I would even be fine with this.
> 

I was counting on the error message to be enough. A dialog box is ok,
there would be a problem to let the user know without running astroid
from the terminal otherwisely.

**Comment by gauteh** [comment]
Gaute Hope writes on January 17, 2016 15:14:
> Stefan Tatschner writes on January 17, 2016 13:04:
>>> Do not crash, but show the user a dialog box saying "Database not configured, RTFM and restart astroid"
>> 
>> I would even be fine with this.
>> 
> 
> I was counting on the error message to be enough. A dialog box is ok,
> there would be a problem to let the user know without running astroid
> from the terminal otherwisely.

We could also use notifications for this. The debug output is a lot more
readable now after the keybindings are not output anymore.

**Comment by samtuke** [comment]
This issue poses a barrier to many new users.

**Comment by gauteh** [comment]
Please test #420.

**Comment by gauteh** [comment]
Fixed in #420.
